pkg/controllers/workapplier/process.go

```diff
@@ -254,7 +253,10 @@ func shouldInitiateTakeOverAttempt(inMemberClusterObj *unstructured.Unstructured
 	// Check if the live object is owned by Fleet.
 	curOwners := inMemberClusterObj.GetOwnerReferences()
 	for idx := range curOwners {
-		if reflect.DeepEqual(curOwners[idx], *expectedAppliedWorkOwnerRef) {
+		if curOwners[idx].UID == expectedAppliedWorkOwnerRef.UID &&
```

Comment:
Hi Britania! The same logic has been repeated for a few times, maybe it's better if we put them in a separate function?